lib/canvas/oauth/missing_scopes_error.rb

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright (C) 2018 - present Instructure, Inc.
+#
+# This file is part of Canvas.
+#
+# Canvas is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, version 3 of the License.
+#
+# Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+module Canvas::Oauth
+  class MissingScopesError < RequestError
+    def initialize(missing_scopes)
+      super("A requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner. "\
+        "The following scopes were requested, but not granted: #{missing_scopes.to_sentence}")
+    end
+
+    def to_render_data
+      {
+        status: 400,
+        json: {
+          error: :missing_scopes,
```

Comment:
One of our devs actually just pointed out that this code change breaks the RFC6749 spec. Note how you've changed the error code from "invalid_scope" to "missing_scopes". It should still be fine that you add the missing scopes to the "error_description" below, but this needs to stay as "invalid_scope".

https://tools.ietf.org/html/rfc6749#section-4.1.2.1

Reply:
It might actually be appropriate to even rename this class InvalidScopeError as well.